presto-main/src/test/java/com/facebook/presto/metadata/TestDiscoveryNodeManager.java

```diff
@@ -244,4 +253,53 @@ public String getPool()
             throw new UnsupportedOperationException();
         }
     }
+
+    private static class InternalNodeWrapper
```

Comment:
While this fixes the test case, wondering if we can have the same equals inside InternalNode so that we can just use InternalNode instead of the wrapper.

Reply:
I thought about this, but since InternalNode has been like this for a long time and is used in critical path, I didn't want to touch it for the test.

Reply:
I think adding equals to InternalNode should be fine , since relying on object reference equality is always a fragile setup and is usually not desired. Do you see any other concerns.?
Also, let us get @tdcmeehan 's views on this.

Reply:
I agree--it really should be safe to add this to `InternalNode`.

Reply:
Sounds good, modified the equals of InternaNode to consider all other properties and removed the wrapper from the test.